src/Goerli-DssSpell.t.sol

```diff
@@ -458,4 +508,494 @@ contract DssSpellTest is GoerliDssSpellTestBase {
+        assertEq(ink, 1 * WAD, "RWA009: bad `ink` after spell"); // Whole unit of collateral is locked
+
+        hevm.warp(block.timestamp + 10 days);
+        jug.drip("RWA009-A");
+
+        (, uint256 rate,,,) = vat.ilks("RWA009-A");
+        assertEq(rate, RAY, 'RWA009: bad `rate`'); // rate keeps being 1 RAY
+
+        // as we have SF 0 we need to pay exectly the same amount of DAI we have drawn
+        uint256 daiToPay = drawAmount;
+
+        // transfer DAI to the URN
+        hevm.store(
+            address(vat),
+            keccak256(abi.encode(address(this), uint256(5))),
+            bytes32(daiToPay * RAY)
+        ); // Forcing extra dai balance for addres(this) on the Vat
+        vat.hope(address(daiJoin));
+        daiJoin.exit(address(this), daiToPay);
+        dai.transfer(address(rwaurn_009), daiToPay);
+        assertEq(dai.balanceOf(address(rwaurn_009)), daiToPay, "Balance of the URN doesnt match");
+
+        // repay debt and free our collateral
+        rwaurn_009.wipe(daiToPay);
+        rwaurn_009.free(1 * WAD);
+
+        // check if MCD_PAUSE_PROXY have RWA009 Tokens
+        assertEq(rwagem_009.balanceOf(address(this)), 1 * WAD, "RWA009: gem not sent back to the caller");
+
+        // check if we have 0 collateral and outstanding debt in the VAT
+        (ink, art) = vat.urns("RWA009-A", address(rwaurn_009));
+        assertEq(ink, 0, "RWA009: bad `ink` after free()");
+        assertEq(art, 0, "RWA009: bad `art` after wipe()");
+    }
+
+    function testRWA009_OPERATOR_LOCK_DRAW_CAGE() public {
+        vote(address(spell));
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        uint256 drawAmount = 25_000_000 * WAD;
+
+        // setting address(this) as operator
+        hevm.store(address(rwaurn_009), keccak256(abi.encode(address(this), uint256(1))), bytes32(uint256(1)));
+        assertEq(rwaurn_009.can(address(this)), 1, "RWA009: address(this) not operator");
+
+        // Check if spell lock 1 * WAD of RWA009
+        assertEq(rwagem_009.balanceOf(address(rwajoin_009)), 1 * WAD, "RWA009: wrong gem balance in join");
+
+        // Check if spell draw 25mm DAI to GENESIS
+        assertEq(dai.balanceOf(address(RWA009_CES_MULTISIG)), drawAmount, "RWA009: wrong dai balance in output conduit");
+
+        (uint256 ink, uint256 art) = vat.urns("RWA009-A", address(rwaurn_009));
+        assertEq(ink, 1 * WAD, "RWA009: wrong ink in urn"); // Whole unit of collateral is locked
+        assertEq(art, drawAmount, "RWA009: wrong art in urn"); // DAI drawn == art as rate should always be 1 RAY
+
+        giveAuth(address(end), address(this));
+        end.cage();
+        end.cage("RWA009-A");
+
+        end.skim("RWA009-A", address(rwaurn_009));
+
+        (ink, art) = vat.urns("RWA009-A", address(rwaurn_009));
+        assertEq(ink, 1 * WAD - (25_000_000 * WAD / 100_000_000), "RWA009: wrong ink in urn after skim");
```

Comment:
Likewise the test for RWA008, this expression can be confusing as it currently stands.

Here we won't have the rounding issue, but we should probably still write the expression the same way:

```
address internal constant RWA009_A_PIP_VALUE = 100_000_000 * WAD;

//...

assertEq(ink, 1 * WAD - ((25_000_000 * WAD) * WAD / RWA009_A_PIP_VALUE), "RWA009: wrong ink in urn after skim");
// ...

Reply:
```suggestion
        uint256 skimmedInk = (25_000_000 * WAD) * WAD / RWA009_PIP_VALUE;
        assertEq(ink, 1 * WAD - skimmedInk, "RWA009: wrong ink in urn after skim");
```

Reply:
Merging for standardization.